nio_cli/commands/config.py

```diff
@@ -26,8 +28,61 @@ def config_project(name='.'):
                 tmp.write('PK_TOKEN={}\n'.format(pk_token))
             else:
                 tmp.write(line)
-        os.remove(env_location)
-        os.rename(tmp.name, env_location)
+        os.remove(conf_location)
+        os.rename(tmp.name, conf_location)
+    secure = input('Optional secure instance configuration [Y/N]: ')
+    if secure.lower() == 'y':
+        config_ssl(name, conf_location)
+
+
+def config_ssl(name, conf_location):
+
+    v = ssl.OPENSSL_VERSION.split()[0]
+    if v != 'OpenSSL':
+        print('No OpenSSL installation detected')
+        return
+
+    ssl_cert = ''
+    ssl_key = ''
+    new_certs = input('Generate a self-signed certificate/key [Y/N]: ')
+
+    if (new_certs.lower() == 'y'):
+        subprocess.call('mkdir ssl &&  cd ssl', shell=True)
+        country = input('Enter two-letter country code: ')
+        state = input('Enter two-letter state code: ')
+        city = input('Enter city: ')
+        org = input('Enter name of organization: ')
+        owner = input('Enter name of owner: ')
+        user = input('Enter name of user: ')
+
+        gen_cert = ('openssl req -newkey rsa:2048 -nodes -keyout key.pem \
```

Comment:
We should probably be generating these certs within Python itself, no need to call a subprocess for it. The `ssl` library will let us create the certs in Python, then we can write them to disk

Reply:
The python docs actually recommend using `openssl` to generate a self-signed cert, so rather than use subprocess calls for the cert and python for the key, I decided to keep it consistent and use subprocess for both. https://docs.python.org/3/library/ssl.html#self-signed-certificates

The `ssl` library lets you create a chain/key/cert but requires a bit more knowledge by the user.  Should the CLI include both options?

Reply:
I'm not sure that documentation is recommending calling with subprocess, I think it's just saying use openSSL which has a Python SDK I believe. Take a look and see if it's doable, in general I want to avoid a bunch of subprocess calls with the CLI if possible. If it's not doable via Python then this will work.